File: ACM/Java/tianti_1/L1_7.java
/*
 * @Author: KasperFan && fanwlx@foxmail.com
 * @Date: 2023-03-04 15:21:34
 * @LastEditTime: 2023-03-17 01:11:01
 * @FilePath: /ACM/Java/tianti_1/L1_7.java
 * @Description: This file is created for learning Code.
 * 
 * Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved. 
 */

package tianti_1;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.StreamTokenizer;

public class L1_7 {
    public static void main(String[] args) throws Exception {
        Read r = new Read();

        int N = r.nextInt(), p, min = (int) 1e6, ansm = 0, max = 0, ansx = 0;
        for (int i = 0; i < N; i++) {
            p = r.nextInt();
            if (min > p) { min = p; ansm = 1;}
            else if (min == p) ansm++;

            if (max < p) { max = p; ansx = 1;}
            else if (max == p) ansx++;
        }
        System.out.println(min + " " + ansm);
        System.out.println(max + " " + ansx);
    }

    /**
     * Read
     */
    static class Read {
        StreamTokenizer st = new StreamTokenizer(new BufferedReader(new InputStreamReader(System.in)));

        public int nextInt() throws Exception {
            st.nextToken();
            return (int) st.nval;
        }
    }
}

File: ACM/Python/tianti_2/L1_7.py
'''
Author: KasperFan && fanwlx@foxmail.com
Date: 2023-03-12 00:29:00
LastEditTime: 2023-03-12 00:45:50
FilePath: /.code/tianti_2/L1_7.py
Description: This file is created for learning Code.
Copyright (c) 2023 by KasperFan in WFU, All Rights Reserved. 
'''

n, m = map(int, input().split())
x = 2**n
lrc = []
for i in range(m):
    lrc.append(input())
mith = []
for i in lrc:
    temp = x
    tex = n-1
    for j in i:
        if j == "y":
            temp -= 2**tex
            tex -= 1
        else:
            tex -= 1
    mith.append(temp)

for i in mith:
    print(i)
